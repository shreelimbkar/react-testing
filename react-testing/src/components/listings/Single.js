/*eslint import/no-unresolved: off*/
import React from 'react';

const Single = ({ item }) =>
  <div className="border w-25 p-3 m-1 mx-auto">
    <img className="card-img-top" src={item.Poster} alt={item.Title} />
    <div className="card-body">
      <a href={item.Website} target="_blank" rel="noopener noreferrer" className="card-title">
        { item.Title }</a>
      <p className="m-0"><small><b>Director:</b> {item.Director}</small></p>
      <p className="m-0"><small><b>Actors:</b> {item.Actors}</small></p>
      <p className="m-0"><small><b>Released:</b> {item.Released}, 
        <b>IMDB Rating:</b> {item.imdbRating}</small>
      </p>
    </div>
  </div>;

export default Single;
