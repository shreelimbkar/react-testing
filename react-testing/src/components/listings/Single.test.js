import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Single from "./Single";

const item = {
    "Actors": "Daniel Radcliffe, Rupert Grint, Emma Watson, Richard Griffiths",
    "Awards": "Nominated for 3 BAFTA Film Awards. Another 11 wins & 42 nominations.",
    "BoxOffice": "$261,835,892",
    "Country": "UK, USA, Germany",
    "DVD": "11 Apr 2003",
    "Director": "Chris Columbus",
    "Genre": "Adventure, Family, Fantasy",
    "Language": "English",
    "Metascore": "63",
    "Plot": "Forced to spend his summer holidays with his muggle relations, Harry Potter gets a real shock when he gets a surprise visitor: Dobby the house-elf, who warns Harry Potter against returning to Hogwarts, for terrible things are going to happen. Harry decides to ignore Dobby's warning and continues with his pre-arranged schedule. But at Hogwarts, strange and terrible things are indeed happening: Harry is suddenly hearing mysterious voices from inside the walls, muggle-born students are being attacked, and a message scrawled on the wall in blood puts everyone on his/her guard - \"The Chamber Of Secrets Has Been Opened. Enemies Of The Heir, Beware\" .",
    "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMTcxODgwMDkxNV5BMl5BanBnXkFtZTYwMDk2MDg3._V1_SX300.jpg",
    "Production": "Warner Bros. Pictures",
    "Rated": "PG",
    "Ratings": [
      {
        "Source": "Internet Movie Database",
        "Value": "7.4/10"
      },
      {
        "Source": "Rotten Tomatoes",
        "Value": "82%"
      },
      {
        "Source": "Metacritic",
        "Value": "63/100"
      }
    ],
    "Released": "15 Nov 2002",
    "Response": "True",
    "Runtime": "161 min",
    "Title": "Harry Potter and the Chamber of Secrets",
    "Type": "movie",
    "Website": "http://harrypotter.warnerbros.com/",
    "Writer": "J.K. Rowling (novel), Steve Kloves (screenplay)",
    "Year": "2002",
    "imdbID": "tt0295297",
    "imdbRating": "7.4",
    "imdbVotes": "444,553"
  };

describe("Single snapshots", () => {
  test("Single Snapshot test", () => {
    const component = renderer.create(<Single item={item} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
