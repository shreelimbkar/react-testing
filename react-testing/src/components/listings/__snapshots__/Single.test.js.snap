// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single snapshots Single Snapshot test 1`] = `
<div
  className="border w-25 p-3 m-1 mx-auto"
>
  <img
    alt="Harry Potter and the Chamber of Secrets"
    className="card-img-top"
    src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTcxODgwMDkxNV5BMl5BanBnXkFtZTYwMDk2MDg3._V1_SX300.jpg"
  />
  <div
    className="card-body"
  >
    <a
      className="card-title"
      href="http://harrypotter.warnerbros.com/"
      target="_blank"
    >
      Harry Potter and the Chamber of Secrets
    </a>
    <p
      className="m-0"
    >
      <small>
        <b>
          Director:
        </b>
         
        Chris Columbus
      </small>
    </p>
    <p
      className="m-0"
    >
      <small>
        <b>
          Actors:
        </b>
         
        Daniel Radcliffe, Rupert Grint, Emma Watson, Richard Griffiths
      </small>
    </p>
    <p
      className="m-0"
    >
      <small>
        <b>
          Released:
        </b>
         
        15 Nov 2002
        , 
        <b>
          IMDB Rating:
        </b>
         
        7.4
      </small>
    </p>
  </div>
</div>
`;
